Add table_aws_ecr_repository. closes #115 (#139)
turbot/steampipe-plugin-aws: docs/tables/aws_ecr_repository.md

```sql
select
  repository_name,
  image_scanning_configuration ->> 'ScanOnPush' as scan_on_push
from
  aws_ecr_repository
where
  image_scanning_configuration ->> 'ScanOnPush' = 'false';
```

### List repositories whose image scanning has failed

```sql
select
  repository_name,
  detail -> 'ImageScanStatus' ->> 'Status' as scan_status
from
  aws_ecr_repository,
  jsonb_array_elements(image_details) as details,
  jsonb(details) as detail
where
  detail -> 'ImageScanStatus' ->> 'Status' = 'FAILED';
```

### List repositories whose tag immutability is disabled

```sql
select
  repository_name,
  image_tag_mutability
from
  aws_ecr_repository
where
  image_tag_mutability = 'IMMUTABLE';
```

### List repositories whose lifecycle policy rule is not configured to remove untagged and old images from your image repository

```sql
select
  repository_name,
  r -> 'selection' ->> 'tagStatus' as tag_status,
  r -> 'selection' ->> 'countType' as count_type
from
  aws_ecr_repository,
  jsonb_array_elements(lifecycle_policy -> 'rules') as r
where
  ((r -> 'selection' ->> 'tagStatus' <>'untagged')
  and (r -> 'selection' ->> 'countType' <>'sinceImagePushed'));

> ```

Reviewer: format the query

# Table: aws_ecr_repository

Amazon Elastic Container Registry (Amazon ECR) is a managed container image registry service.

## Examples

### SSM association basic info
```sql
select
  repository_name,
  registry_id,
  repository_arn,
  repository_uri,
  created_at,
  region,
  account_id
from
  aws_ecr_repository;
```

> ### List of ECR repositories which are not using Customer Managed Keys(CMK) for encryption

Reviewer: ```suggestion
### List of ECR repositories which are not using Customer Managed Keys(CMK) for encryption

```

```sql
select
  repository_name,
  encryption_configuration ->> 'EncryptionType' as encryption_type,
  encryption_configuration ->> 'KmsKey' as kms_key
from
  aws_ecr_repository
where
  encryption_configuration ->> 'KmsKey' = 'KMS';
```

> ### List Repositories whose imageTag set as MUTABLE

Reviewer: ```suggestion
### List Repositories whose imageTag set as MUTABLE

```
Add this space in all the query

```sql
select
  repository_name,
  image_tag_mutability
from
  aws_ecr_repository
where
  image_tag_mutability = 'MUTABLE';
```

> 

Reviewer: Add a new query `List of repositories where automatic image scanning is disable`

### Count of image details in each Repository
```sql
select
  count(image_details) as count
from
  aws_ecr_repository
where
  repository_name = 'steampipe-12';

> ```

Reviewer: Is the above query working? `image_details` is an array

Author: yes it's working.

# Table: aws_ecr_repository

Amazon Elastic Container Registry (Amazon ECR) is a managed container image registry service.

## Examples

### SSM association basic info

```sql
select
	repository_name,
	registry_id,
	repository_arn,
	repository_uri,
	created_at,
	region,
	account_id
from
	aws_ecr_repository;

> ```

Reviewer: ```
select
  repository_name,
  registry_id,
  repository_arn,
  repository_uri,
  created_at,
  region,
  account_id
from
  aws_ecr_repository;
```

Reviewer: Fix indentation for all the queries.

### List Repositories whose imageTag set as MUTABLE

```sql
select
	repository_name,
	image_tag_mutability
from
	aws_ecr_repository
where
	image_tag_mutability = 'MUTABLE';
```

> 

Reviewer: ```suggestion

```

> ### List Repositories who has image scan configuration with both CRITICAL findings 

Reviewer: the heading is not appropriate

```sql
select
  repository_name,
	image_scanning_configuration as scan
from
	aws_ecr_repository
where

> 	finding-severity-counts= 'CRITICAL';

Reviewer: I can not find any column `finding-severity-counts` by this name

> ```

Reviewer: ```suggestion
```

```

> ```

Reviewer: ```suggestion
```

```

> ### Count of image ids in a requested Repository

Reviewer: ```suggestion

### Count of image ids in a requested Repository
```

```sql
select
	count(image_ids) as count
from
	aws_ecr_repository
where
	repository_name = 'steampipe-12';

> ```

Reviewer: The above query is not working. Rewrite the query according to the heading

> ### List of unencrypted ECR repository

Reviewer: ```suggestion

### List of unencrypted ECR repository
```

```sql
select
	repository_name,
	encryption_configuration ->> 'EncryptionType' as encryption_type,
	encryption_configuration ->> 'KmsKey' as kms_key
from
	aws_ecr_repository
where
	encryption_configuration ->> 'KmsKey' = 'null';
```

> 

Reviewer: ```suggestion

```

> ### List of ECR repository which are not using Customer Managed Keys(CMK)

Reviewer: ```suggestion
### List of ECR repositories which are not using Customer Managed Keys(CMK) for encryption
```

```sql
select
	repository_name,
	encryption_configuration ->> 'EncryptionType' as encryption_type,
	encryption_configuration ->> 'KmsKey' as kms_key
from
	aws_ecr_repository
where
	encryption_configuration ->> 'KmsKey' = 'KMS';

> ```

Reviewer: The below 2 queries should be the 2nd and 3rd query respectively

# Table: aws_ecr_repository

Amazon Elastic Container Registry (Amazon ECR) is a managed container image registry service.

## Examples

### SSM association basic info

```sql
select
  repository_name,
  registry_id,
  repository_arn,
  repository_uri,
  created_at,
  region,
  account_id
from
  aws_ecr_repository;
```

### List of ECR repositories which are not using Customer Managed Keys(CMK) for encryption

```sql
select
  repository_name,
  encryption_configuration ->> 'EncryptionType' as encryption_type,
  encryption_configuration ->> 'KmsKey' as kms_key
from
  aws_ecr_repository
where
  encryption_configuration ->> 'EncryptionType' = 'AES256';
```

### List Repositories whose tag immutability is disabled

```sql
select
  repository_name,
  image_tag_mutability
from
  aws_ecr_repository
where
  image_tag_mutability = 'IMMUTABLE';
```

### List of repositories where automatic image scanning is disabled

```sql
select
  repository_name,
  image_scanning_configuration ->> 'ScanOnPush' as scan_on_push
from
  aws_ecr_repository
where
  image_scanning_configuration ->> 'ScanOnPush' = 'false';
```

### Image vulnerability count by severity. 

```sql
select
  repository_name,

>   detail -> 'ImageScanFindingsSummary' -> 'FindingSeverityCounts' -> 'INFORMATIONAL' as severity_counts

Reviewer: ```suggestion
  detail -> 'ImageScanFindingsSummary' -> 'FindingSeverityCounts' -> 'INFORMATIONAL' as informational_severity_counts
```

Reviewer: show the low and the medium severity count too

# Table: aws_ecr_repository

Amazon Elastic Container Registry (Amazon ECR) is a managed container image registry service.

## Examples

> ### ECR repository basic info

Reviewer: ```suggestion
### Basic info
```

```sql
select
  repository_name,
  registry_id,
  arn,
  repository_uri,
  created_at,
  region,
  account_id
from
  aws_ecr_repository;
```

### List of ECR repositories which are not using Customer Managed Keys(CMK) for encryption

```sql
select
  repository_name,
  encryption_configuration ->> 'EncryptionType' as encryption_type,
  encryption_configuration ->> 'KmsKey' as kms_key
from
  aws_ecr_repository
where
  encryption_configuration ->> 'EncryptionType' = 'AES256';
```

> ### List repositories where automatic image scanning is disabled

Reviewer: ```suggestion
### List repositories whose automatic image scanning is disabled
```

  created_at,
  region,
  account_id
from
  aws_ecr_repository;
```

### List of ECR repositories which are not using Customer Managed Keys(CMK) for encryption

```sql
select
  repository_name,
  encryption_configuration ->> 'EncryptionType' as encryption_type,
  encryption_configuration ->> 'KmsKey' as kms_key
from
  aws_ecr_repository
where
  encryption_configuration ->> 'EncryptionType' = 'AES256';
```

### List of repositories where automatic image scanning is disabled

```sql
select
  repository_name,
  image_scanning_configuration ->> 'ScanOnPush' as scan_on_push
from
  aws_ecr_repository
where
  image_scanning_configuration ->> 'ScanOnPush' = 'false';
```

### List of repositories whose image scanning has failed

```sql
select
  repository_name,
  detail -> 'ImageScanStatus' ->> 'Status' as scan_status
from
  aws_ecr_repository,
  jsonb_array_elements(image_details) as details,
  jsonb(details) as detail
where
  detail -> 'ImageScanStatus' ->> 'Status' = 'FAILED';
```

### List Repositories whose tag immutability is disabled

```sql
select
  repository_name,
  image_tag_mutability
from
  aws_ecr_repository
where
  image_tag_mutability = 'IMMUTABLE';
```

> ### Repository image vulnerability count by severity 

Reviewer: ```suggestion
### Get repository image vulnerability count by severity 
```
